coverage-reporter/build.gradle.kts

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023
+
+plugins {
+    kotlin("jvm") version "1.9.21"
+    `maven-publish`
+    idea
+    `java-gradle-plugin`
+    id("signing")
+}
+
+group = "io.github.amosproj"
+version = "1.0"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.jetbrains.kotlin:kotlin-test")
+    // https://mvnrepository.com/artifact/org.pitest/pitest-command-line
+    implementation("org.pitest:pitest-command-line:1.15.3")
+}
+
+tasks.test {
+    useJUnitPlatform()
+}
+
+kotlin {
+    jvmToolchain(17)
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            groupId = group.toString()
+            artifactId = "coverage-reporter"
+            version = project.version.toString()
+            from(components["java"])
+
+            pom {
+                name.set("Coverage Reporter")
+                description.set("Coverage Report writer for PIT")
+            }
+        }
+    }
+
+    repositories {
+        maven {
+            credentials {
+                username = project.findProperty("spaceUsername").toString()
+                password = project.findProperty("spacePassword").toString()
+            }
+            url = uri("https://maven.pkg.jetbrains.space/pitmutationmate/p/main/coverage-reporter")
+        }
+    }
```

Comment:
[This](https://www.jetbrains.com/help/space/consume-maven-artifacts.html) in step 5 reads like one would have to always specify the repository creds even as a user?
Or am i understanding this wrong?
It sounds like a private repository

Reply:
Yes exactly. So the way I implemented here is that you need the credentials stored in the gradle.properties file, which is not uploaded to the repo.

Reply:
Yeah i get that but the way I understand the documentation of this repository is that it is entirely **private** as in i could only download the coverage listener if i had the credentials. So no one can use this without them. Not only publish. Thats what it sounds like to me at least from the documentation

Reply:
Ah okay I see your point. Actually I have to publish it here https://s01.oss.sonatype.org. I also changed that locally. Once I can use the coverage-reporter through maven-central I will upload my changes.